### DISCUSSION TITLE
Exporting Order using XSD file

### DISCUSSION THREAD

**Discussion by nimitshukla123** [SalesforceCommerceCloud/sfcc-ci#545] [Team Posts]
HI All

i have a requirement in which i need to fetch all the attributes of an order according to given XSD file and create an XML.

Just need to know if there is any API which can do this or we have to read the XSD file and iterate out the elements and then create XML.

Thanks

**Comment by pravngaur** [comment]
Hi Nimit,

I do not think there is any api which generates xml files inline to any
other xsd apart from the platform's one. I will look into documentation
again, and write back.

But most probably, there isn't any api and you would need to post process
the xml files, once they are exported from SFCC. More importantly, i would
suggest you to do it at ETL layer not in commerce engine, doing file
transformation on SFCC might give you performance issues.

If you don't have ETL/middleware in your architecture then create a thin
utility outside of commerce and do the transformation there.

**Comment by hnestmann** [comment]
Hi Nimit, 
If you want a representaion of the order based on the xsd, why don't you just export the order?

But you have APIs like .describe() on objects which allow you to understand their meta model. Also you can setup a job, which export the system type definitions as part of site-export.
